governance/contract.go

```diff
@@ -80,41 +77,50 @@ func (e *ContractEngine) UpdateParams() error {
 
 // contractGetAllParamsAt sets evmCtx.BlockNumber as num
 func (e *ContractEngine) contractGetAllParamsAt(num uint64) (*params.GovParamSet, error) {
-	if e.chain == nil {
-		logger.Info("ContractEngine disabled: chain = nil")
-		return params.NewGovParamSet(), nil
+	chain := e.headerGov.BlockChain()
+	if chain == nil {
+		logger.Crit("headerGov.BlockChain() is nil")
+		return nil, errContractEngineNotReady
 	}
 
-	if !e.chainConfig.IsKoreForkEnabled(e.chain.CurrentHeader().Number) {
+	config := chain.Config()
+	if !config.IsKoreForkEnabled(new(big.Int).SetUint64(num)) {
 		logger.Info("ContractEngine disabled: hardfork block not passed")
 		return params.NewGovParamSet(), nil
 	}
 
-	addr, err := e.contractAddrAt(num)
-	if err != nil {
-		return nil, err
-	}
-
+	addr := e.contractAddrAt(num)
 	if common.EmptyAddress(addr) {
 		logger.Info("ContractEngine disabled: GovParamContract address not set")
```

Comment:
If `contractAddrAt` returns `common.Address{}`, it seems to be an error in some cases. Shouldn't it be necessary to distinguish between an error case and a non-error case?

Reply:
@yoomee1313  I think it got overwritten during rebase. Thanks!

Reply:
fixed in 03d4270b